db_processor.bal

```diff
@@ -303,13 +330,32 @@ function addAidPackageItem(AidPackageItem aidPackageItem) returns int|error {
 }
 
 function constructAidPAckageItem(int packageId, AidPackageItem aidPackageItem) returns error? {
+    string errorMessage = "";
     Quotation quotation = check getQuotation(aidPackageItem.quotationID);
     aidPackageItem.quotation = quotation;
     aidPackageItem.packageID = packageId;
     aidPackageItem.totalAmount = <decimal>aidPackageItem.quantity * quotation.unitPrice;
     aidPackageItem.packageItemID = check addAidPackageItem(aidPackageItem);
-    check updateMedicalNeedQuantity(aidPackageItem.needID);
-    check updateQuotationRemainingQuantity(aidPackageItem);
+    transaction {
+        error? res =  updateMedicalNeedQuantity(aidPackageItem.needID);
+        if (res is error) {
+            errorMessage += generateTransactionErrorMessage(res, update);
+        }
+
+        res = updateQuotationRemainingQuantity(aidPackageItem);
+        if (res is error) {
+            errorMessage += generateTransactionErrorMessage(res, update);
+        }
+        if (errorMessage != "") {
+            rollback;
+            return error(errorMessage);
+        } else {
+            error? err = commit;
+            if err is error {
+                return error("Error occurred while committing, ", err);
```

Comment:
Isn't it better to say error while committing what? Something like below and a similar meaningful message for all the other places?

```
return error("Error occurred while committing aid package item, ", err);
```

Reply:
Will update.

Reply:
Updated.
Please check.